py/server/deephaven/experimental/s3.py

```diff
@@ -84,6 +89,28 @@ def __init__(self,
             num_concurrent_write_parts (int): the maximum number of parts that can be uploaded concurrently when writing
                 to S3 without blocking, defaults to 64. Setting a higher value may increase throughput, but may also
                 increase memory usage.
+            profile_name (str): the default profile name used for configuring the default region, credentials, etc.,
+                when reading or writing to S3. If not provided, the AWS SDK picks the profile name from the
+                'aws.profile' system property, the "AWS_PROFILE" environment variable, or defaults to "default".
+                Setting a profile name assumes that the credentials are provided via this profile; if that is not the
+                case, you must explicitly set credentials.
+            config_file_path (str): the path to the configuration file to use for configuring the default region,
+                credentials, etc. when reading or writing to S3. If not provided, the AWS SDK picks the configuration
+                file from the 'aws.configFile' system property, the "AWS_CONFIG_FILE" environment variable, or defaults
+                to "{user.home}/.aws/config".
+                Setting a configuration file path assumes that the credentials are provided via the config and
+                credentials files; if that is not the case, you must explicitly set credentials.
+            credentials_file_path (str): the path to the credentials file to use for configuring the default region,
+                credentials, etc. when reading or writing to S3. If not provided, the AWS SDK picks the credentials file
+                from the 'aws.credentialsFile' system property, the "AWS_CREDENTIALS_FILE" environment variable, or
+                defaults to "{user.home}/.aws/credentials".
+                Setting a credentials file path assumes that the credentials are provided via the config and
+                credentials files; if that is not the case, you must explicitly set credentials.
+            profile_credentials (bool): use the profile name to load the credentials from the config and credentials
+                file and fail if none found. Default is False. Cannot be combined with other credentials.
```

Comment:
1. would `use_profile_credentials_` or `use_profile_creds_`  be more accurate?
2. Not clear about `Cannot be combined with other credentials`